lib/amqp/channel/direct_consumer.ex

```diff
@@ -0,0 +1,139 @@
+defmodule AMQP.DirectConsumer do
+  @moduledoc """
+    Direct consumer callback module that implements `:amqp_gen_consumer` behavior. This allows the process calling
+    `AMQP.Channel.open/2` to receive messages directly from the channel process, skipping intermediate processes.
```

Comment:
I don't think that it describes the purpose of module well. Can you rewrite please?

Reply:
It will be great if you can put an example(usage) too